core/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -467,10 +468,17 @@ protected void doXContentAnalyzers(XContentBuilder builder, boolean includeDefau
             if (includeDefaults) {
                 builder.field("analyzer", "default");
             }
-        } else if (includeDefaults || fieldType().indexAnalyzer().name().startsWith("_") == false && fieldType().indexAnalyzer().name().equals("default") == false) {
-            builder.field("analyzer", fieldType().indexAnalyzer().name());
-            if (fieldType().searchAnalyzer().name().equals(fieldType().indexAnalyzer().name()) == false) {
+        } else {
+            boolean writtenSearchAnalyzer = false;
+            if (includeDefaults || fieldType().searchAnalyzer().name().startsWith("_") == false
+                    && fieldType().searchAnalyzer().name().equals("default") == false
+                    && fieldType().searchAnalyzer().name().equals(fieldType().indexAnalyzer().name()) == false) {
                 builder.field("search_analyzer", fieldType().searchAnalyzer().name());
+                writtenSearchAnalyzer = true;
+            }
+            if (writtenSearchAnalyzer || includeDefaults || fieldType().indexAnalyzer().name().startsWith("_") == false
+                    && fieldType().indexAnalyzer().name().equals("default") == false) {
+                builder.field("analyzer", fieldType().indexAnalyzer().name());
```

Comment:
@rjernst I have changed this so if the `search_analyzer` is written we bypass the checks on whether to write the `analyzer` since we want to write the output in the "new format" where `search_analyzer` cannot be specified without `analyzer`. However, I can't see a way of simplifying the other checks since we need to check whether the search_analyzer is `default` or starts with `_` before we write it. Let me know if you can see a better way to write this.